src/mongo/mongoConnectionStrings.ts

```diff
@@ -63,22 +63,19 @@ export async function parseMongoConnectionString(connectionString: string): Prom
         }
     }
 
-    const serverConfig: Server | ReplSet | Mongos = mongoClient.db(testDb).serverConfig;
-    // Azure CosmosDB comes back as a ReplSet
-    if (serverConfig instanceof ReplSet) {
-        // get the first connection string from the servers for the ReplSet
-        // this may not be best solution, but the connection (below) gives
-        // the replicaset host name, which is different than what is in the connection string
-        // "s" is not part of ReplSet static definition but can't find any official documentation on it. Yet it is definitely there at runtime. Grandfathering in.
-        // tslint:disable-next-line:no-any
-        const rs: any = serverConfig;
-        host = rs.s.options.servers[0].host;
-        port = rs.s.options.servers[0].port;
+    // tslint:disable-next-line:no-any
+    const serverConfig: any = mongoClient.db(testDb).serverConfig;
+
+    // get the first connection string from the servers list
+    // this may not be best solution, but the connection (below) gives
+    // host name of single server, mongos instance or the primany from replicaSet which is different than what is in the connection string (espcially for Replica sets)
+    // "s" is not part of the static definition but can't find any official documentation on it. Yet it is definitely there at runtime. Grandfathering in.
+    if (serverConfig.s) {
```

Comment:
Because this is type `any` and because we're close to a release, I think we want to be even more careful

something like this:
```
const host: string = rs?.s?.options?.servers[0]?.host || serverConfig.host;
const port: string = rs?.s?.options?.servers[0]?.port || serverConfig.port;
```

Reply:
I see what you mean. I agree, it is better to use both values.